### PR TITLE
OCPBUGS-24557: Explain spec.clusterImageSetNameRef and spec.clusters.clusterImageSetNameRef in SiteConfig CR

### PR DESCRIPTION
**Pull request by amolnar-gh**
Version(s): 4.12, 4.13, 4.14, 4.15, 4.16

Issue: https://issues.redhat.com/browse/OCPBUGS-24557

Link to docs preview: 
* [Single-node OpenShift SiteConfig CR installation reference](https://71724--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-sites#ztp-sno-siteconfig-config-reference_ztp-deploying-far-edge-sites)
* [Pre-caching images in GitOps ZTP](https://71724--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-precaching-tool#ztp-pre-caching-config-con_pre-caching)

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/ztp-precaching-ztp-config.adoc [hunk 1]
```diff
@@ -75,6 +75,8 @@ spec:
             - name: "ens1f0"
               macAddress: "AA:BB:CC:11:22:33"
 ----
+<1> References the cluster image set that is used for deployment, unless you specify a different image set in the `spec.cluster.clusterImageSetNameRef` field.
```
Comment (sabbir-47) [inline modules/ztp-precaching-ztp-config.adoc:78]: nit. clusters and not cluster

**Comment by mpmaruthu** [comment]
Telco RAN QE: Looks good to me.
/Peri M

File: modules/ztp-precaching-ztp-config.adoc [hunk 2]
```diff
@@ -75,6 +75,8 @@ spec:
             - name: "ens1f0"
               macAddress: "AA:BB:CC:11:22:33"
 ----
+<1> References the cluster image set that is used for deployment, unless you specify a different image set in the `spec.clusters.clusterImageSetNameRef` field.
```
Comment (bburt-rh) [inline modules/ztp-precaching-ztp-config.adoc:78]:
```suggestion
<1> Specifies the cluster image set used for deployment, unless you specify a different image set in the `spec.clusters.clusterImageSetNameRef` field.
```
- Would `specifies` be more accurate than `references` here?
- Edit for minimalism

File: modules/ztp-precaching-ztp-config.adoc [hunk 3]
```diff
@@ -75,6 +75,8 @@ spec:
...
+<2> References the cluster image set that is used to deploy an individual cluster. If defined, it overrides the `spec.clusterImageSetNameRef` at site level.
```
Comment (bburt-rh) [inline modules/ztp-precaching-ztp-config.adoc:79]:
```suggestion
<2> Specifies the cluster image set used to deploy an individual cluster. If defined, it overrides the `spec.clusterImageSetNameRef` at the site level.
```
- Would `specifies` be more accurate than `references` here?
- Edit for minimalism

File: modules/ztp-sno-siteconfig-config-reference.adoc [hunk 4]
```diff
@@ -36,6 +36,9 @@ Use the reference configuration as specified in the example `SiteConfig` CR.
 Adding additional components back into the system might require additional reserved CPU capacity.
 ====
 
+|`spec.clusters.clusterImageSetNameRef`
+|References the cluster image set that is used to deploy an individual cluster. If defined, it overrides the `spec.clusterImageSetNameRef` at site level.
```
Comment (bburt-rh) [inline modules/ztp-sno-siteconfig-config-reference.adoc:40]:
```suggestion
|Specifies the cluster image set used to deploy an individual cluster. If defined, it overrides the `spec.clusterImageSetNameRef` at the site level.
```
- Same comment as above re `specifies` vs `references`
- Edit for minimalism

**Review by bburt-rh — APPROVED** [review]
I added a couple of suggestions for you to consider, but otherwise LGTM. Thank you!

**Comment by bburt-rh** [comment]
Also, if this change is going into the `main` branch, it should be for 4.16, too, right? Can you please update the description to add 4.16?

**Review by skrthomas — COMMENTED** [review]
Merge review lgtm

**Pull request merged**